### PR TITLE
Issues/29 add record state

### PR DESCRIPTION
**Pull request by kostobog**
#29 - Implement generic record state.

### CODE REVIEW

File: src/main/resources/model.ttl [hunk 1]
```diff
@@ -163,5 +167,28 @@ rm:user rdf:type owl:Class ;
 rm:impersonator rdf:type owl:Class ;
         rdfs:label "Impersonator"@en .
 
+###  http://onto.fel.cvut.cz/ontologies/record-manager/record-state
+rm:record-state rdf:type owl:Class ;
+        rdfs:label "record state"@en .
+
+###  http://onto.fel.cvut.cz/ontologies/record-manager/open-record-state
+rm:open-record-state rdf:type owl:Class ;
```
[the remark this answers is not in the document]
Comment (kostobog) [reply src/main/resources/model.ttl:175]: @blcham 
I understand. After some considerations I propose to have `rm:has-state` as sub property of `rdf:type`:

```
###  http://onto.fel.cvut.cz/ontologies/record-manager/has-state
rm:has-state rdf:type owl:ObjectProperty ;
                rdfs:subPropertyOf rdf:type ;
                rdfs:label "has state"@en .
...

###  http://onto.fel.cvut.cz/ontologies/record-manager/record-state
rm:record-state rdf:type owl:Class ;
        rdfs:subClassOf ufo:phase ;
        rdfs:label "record state"@en .
```

File: src/main/resources/model.ttl [hunk 2]
```diff
@@ -74,6 +74,10 @@ rm:relates-to rdf:type owl:ObjectProperty .
 rm:was-treated-at rdf:type owl:ObjectProperty ;
                   rdfs:subPropertyOf rm:relates-to .
 
+###  http://onto.fel.cvut.cz/ontologies/record-manager/has-state
+rm:has-state rdf:type owl:ObjectProperty ;
+                rdfs:label "has-state"@en .
```
[the remark this answers is not in the document]
Comment (kostobog) [reply src/main/resources/model.ttl]: Fixed

**Review by blcham — APPROVED** [review]
LGTM, although i suggest to make it consistent ... if you started to call it phase, refactor all occurences of state

**Comment by LaChope** [comment]
@blcham 
There are still issues, when posting the "phase" to backend, it does not return it and do not save it to database, see below:

image

image

**Comment by blcham** [comment]
@LaChope try to search "state" instead "phase" ... because @kostobog did not refactor actual RDF model  behind it

File: src/main/java/cz/cvut/kbss/study/model/PatientRecord.java [hunk 3]
```diff
@@ -52,6 +45,10 @@ public class PatientRecord implements Serializable, HasOwlKey {
             CascadeType.REMOVE}, fetch = FetchType.EAGER)
     private Question question;
 
+    @Enumerated(EnumType.OBJECT_ONE_OF)
+    @OWLObjectProperty(iri = Vocabulary.s_p_has_state)
```
[the remark this answers is not in the document]
Comment (kostobog) [reply src/main/java/cz/cvut/kbss/study/model/PatientRecord.java]: Refactored in [164974d](https://github.com/kbss-cvut/record-manager/pull/31/commits/164974dc726d8e9d3ce3ac1db8534be5c30cbafb) and [4ae185f](https://github.com/kbss-cvut/record-manager/pull/31/commits/4ae185fde400cf8f32c5ae72497728fb4369f1df)

**Comment by blcham** [comment]
@kostobog in what state is this PR?

**Review by LaChope — APPROVED** [review]

**Comment by LaChope** [comment]
Everything works now, LGTM

**Pull request merged**